### PR TITLE
Support Ubuntu >= 22.04 and Debian >= 11

### LINKED ISSUE #95: Support ubuntu 22.04
## Problem Statement

Ubuntu 22.04 doesn't have the [`python-openssl`](https://packages.ubuntu.com/search?keywords=python-openssl&searchon=names) package anymore, and it's called `python3-openssl` now

## Possible Solution

Install `python3-openssl` instead of `python-openssl` on ubuntu 22.04

### PR DESCRIPTION
**Pull request by gbagnoli**
Signed-off-by: Giacomo Bagnoli <giacomo@bagnoli.me>

# Description

Add support for Ubuntu 22.04 by installing `python3-openssl` instead of the old name `python-openssl`
Also added Ubuntu 22.04 to the tests kitchen files

## Issues Resolved

Fixes #95

## Check List

- [X] A summary of changes made is included in the CHANGELOG under `## Unreleased`
- [X] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.

### CODE REVIEW

**Comment by gbagnoli** [comment]
Not sure how to add results from kitchen tests but:

22.04m user install
```
Profile Summary: 6 successful controls, 0 control failures, 0 controls skipped
Test Summary: 34 successful, 0 failures, 0 skipped
       Finished verifying <user-install-ubuntu-2204> (0m3.47s).
-----> Test Kitchen is finished. (4m58.47s)
```

22.04 system install
```
Profile Summary: 4 successful controls, 0 control failures, 0 controls skipped
Test Summary: 25 successful, 0 failures, 0 skipped
       Finished verifying <system-install-ubuntu-2204> (0m3.20s).
```

20.04 user install
```
Profile Summary: 6 successful controls, 0 control failures, 0 controls skipped
Test Summary: 34 successful, 0 failures, 0 skipped
       Finished verifying <user-install-ubuntu-2204> (0m3.47s).
-----> Test Kitchen is finished. (4m58.47s)
```

File: libraries/helpers.rb [hunk 1]
```diff
@@ -53,7 +53,11 @@ def pip_command(command)
       def pyenv_prerequisites
         case node['platform_family']
         when 'debian'
-          %w(make libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev xz-utils tk-dev libffi-dev liblzma-dev python-openssl git)
+          %w(make libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev xz-utils tk-dev libffi-dev liblzma-dev git) << if platform?('ubuntu') && node['lsb']['codename'] == 'jammy'
```
Comment (Stromweld) [inline libraries/helpers.rb:56]: You shouldn’t be comparing the code name. This isn’t future proof and will be broken with next release of Ubuntu even if there is no change. You should be comparing the platform version. This also is specific to Ubuntu when typically Debian and Ubuntu share the same package names. Have you verified Debian’s latest version isn’t using python3-OpenSSL as well?

**Review by Stromweld — CHANGES_REQUESTED** [review]

File: libraries/helpers.rb (hunk 1 above)
Comment (gbagnoli) [reply libraries/helpers.rb:56]: thanks for the comment. Yeah, it makes more sense checking the version.

you are also right about debian - bullseye and on (last stable) seems to not have python-openssl as well (https://packages.debian.org/bullseye/python-openssl vs https://packages.debian.org/bullseye/python3-openssl)

python-openssl exists up to buster https://packages.debian.org/buster/python-openssl

I'll make the changes.

**Comment by gbagnoli** [comment]
```
chef exec kitchen list
Instance                     Driver  Provisioner  Verifier  Transport  Last Action  Last Error
system-install-debian-9      Dokken  Dokken       Inspec    Dokken     Verified     <None>
system-install-debian-10     Dokken  Dokken       Inspec    Dokken     Verified     <None>
system-install-debian-11     Dokken  Dokken       Inspec    Dokken     Verified     <None>
system-install-centos-7      Dokken  Dokken       Inspec    Dokken     Verified     <None>
system-install-centos-8      Dokken  Dokken       Inspec    Dokken     Verified     <None>
system-install-amazon-linux  Dokken  Dokken       Inspec    Dokken     Verified     <None>
system-install-oracle-linux  Dokken  Dokken       Inspec    Dokken     Verified     <None>
system-install-fedora        Dokken  Dokken       Inspec    Dokken     Verified     <None>
system-install-ubuntu-1804   Dokken  Dokken       Inspec    Dokken     Verified     <None>
system-install-ubuntu-2004   Dokken  Dokken       Inspec    Dokken     Verified     <None>
system-install-ubuntu-2204   Dokken  Dokken       Inspec    Dokken     Verified     <None>
user-install-debian-9        Dokken  Dokken       Inspec    Dokken     Verified     <None>
user-install-debian-10       Dokken  Dokken       Inspec    Dokken     Verified     <None>
user-install-debian-11       Dokken  Dokken       Inspec    Dokken     Verified     <None>
user-install-centos-7        Dokken  Dokken       Inspec    Dokken     Verified     <None>
user-install-centos-8        Dokken  Dokken       Inspec    Dokken     Verified     <None>
user-install-amazon-linux    Dokken  Dokken       Inspec    Dokken     Verified     <None>
user-install-oracle-linux    Dokken  Dokken       Inspec    Dokken     Verified     <None>
user-install-fedora          Dokken  Dokken       Inspec    Dokken     Verified     <None>
user-install-ubuntu-1804     Dokken  Dokken       Inspec    Dokken     Verified     <None>
user-install-ubuntu-2004     Dokken  Dokken       Inspec    Dokken     Verified     <None>
user-install-ubuntu-2204     Dokken  Dokken       Inspec    Dokken     Verified     <None>

```

**Comment by Stromweld** [comment]
Can you also update the .github/workflows/ci.yaml file to look similar to this one https://github.com/sous-chefs/selinux/blob/main/.github/workflows/ci.yml. That'll fix the pipeline errors and let us see the integration tests here as well. Thanks.

**Review by Stromweld — CHANGES_REQUESTED** [review]

**Comment by gbagnoli** [comment]
@Stromweld - I've updated the CI, fixed the lint and the changelog. Do you want me to squash and republish?

**Review by Stromweld — APPROVED** [review]
Looks good. Thanks for the PR. We can squash the commits on merge in github here.

**Comment by kitchen-porter** [comment]
Released as: [4.2.0](https://github.com/sous-chefs/pyenv/releases/tag/4.2.0)

**Pull request merged**